src/dex/uniswap-v2/dystopia-stable.ts

```diff
@@ -0,0 +1,248 @@
+  private static SWAP_FEE_FACTOR: bigint = BigInt(2000);
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(DystopiaStableConfig);
+
+  constructor(
+    protected network: Network,
+    protected dexKey: string,
+    protected dexHelper: IDexHelper,
+  ) {
+    super(
+      network,
+      dexKey,
+      dexHelper,
+      false,
+      DystopiaStableConfig[dexKey][network].factoryAddress,
+      DystopiaStableConfig[dexKey][network].subgraphURL,
+      DystopiaStableConfig[dexKey][network].initCode,
+      DystopiaStableConfig[dexKey][network].feeCode,
+      DystopiaStableConfig[dexKey][network].poolGasCost,
+    );
+
+    // swaps on Dystopia should use own router
+    const router = DystopiaStableConfig[dexKey][network].router;
+    if (router) this.router = router;
+
+    this.factory = new dexHelper.web3Provider.eth.Contract(
+      dystopiaFactoryABI as any,
+      DystopiaStableConfig[dexKey][network].factoryAddress,
+    );
+  }
+
+  async findPair(from: Token, to: Token) {
+    if (from.address.toLowerCase() === to.address.toLowerCase()) return null;
+    const [token0, token1] =
+      from.address.toLowerCase() < to.address.toLowerCase()
+        ? [from, to]
+        : [to, from];
+
+    const key = `${token0.address.toLowerCase()}-${token1.address.toLowerCase()}`;
+    let pair = this.pairs[key];
+    if (pair) return pair;
+    // try to find non-stable pair first
+    let exchange = await this.factory.methods
+      // Dystopia has additional boolean parameter "StablePool"
+      // At this DystopiaStable implementation we're looking for
+      // stable (x3y+y3x) pools only
+      .getPair(token0.address, token1.address, true)
+      .call();
+
+    if (exchange === NULL_ADDRESS) {
+      pair = { token0, token1 };
+    } else {
+      pair = { token0, token1, exchange };
+    }
+    this.pairs[key] = pair;
+    return pair;
+  }
+
+  async getSellPrice(
+    priceParams: UniswapV2PoolOrderedParamsWithDecimals,
+    srcAmount: bigint,
+  ): Promise<bigint> {
+    const { reservesIn, reservesOut, decimalsIn, decimalsOut } = priceParams;
+
+    if (BigInt(reservesIn) + srcAmount > RESERVE_LIMIT) {
+      return 0n;
+    }
+
+    const amountIn = srcAmount - srcAmount / DystopiaStable.SWAP_FEE_FACTOR;
+
+    const reservesInN = BigInt(reservesIn);
+    const reservesOutN = BigInt(reservesOut);
+    const decimalsInN = BigInt(decimalsIn);
+    const decimalsOutN = BigInt(decimalsOut);
+
+    const xy = _k(reservesInN, reservesOutN, decimalsInN, decimalsOutN);
+    const reserveA = (reservesInN * e18) / decimalsInN;
+    const reserveB = (reservesOutN * e18) / decimalsOutN;
+    const amountInNorm = (amountIn * e18) / decimalsInN;
+    const y = reserveB - _getY(amountInNorm + reserveA, xy, reserveB);
+    const returnAmount = (y * decimalsOutN) / e18;
+    return returnAmount;
+  }
+
+  // TODO Do we have to implement getBuyPrice? (buy side do not used in tests)
```

Comment:
Can we omit buy side for now and integrate sell only? 

As I have no ready formula for stable pools swaps and it can take some time to properly made it, and we want to integrate  ASAP. 

I guess sell side is used in most cases..

Reply:
Ok, I discussed this request with @erez-paraswap and received his approval. We can integrate first with `SELL` and `BUY` later.

As for how is it better to do, please check, for example `Nerve`:
https://github.com/paraswap/paraswap-dex-lib/blob/f5d7319fc6cc3cba7c888b2b18027dd41e891328/src/dex/nerve/nerve.ts#L172

In general: for `getPricesVolume` and `getPoolIfentifiers` you check the side and return `[]`, as for building params, you should throw an `Error` in `getSimpleParam` and `getAdapterParam`.